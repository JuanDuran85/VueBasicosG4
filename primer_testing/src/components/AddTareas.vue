<template>
    <div class="container my-5">
        <h2 class="text-center my-4">Agrega una Actividad por Realizar</h2>
        <div class="input-group mb-3">
            <input type="text" class="form-control" placeholder="Agrega una actividad" aria-label="Agrega una actividad" aria-describedby="button-addon2" v-model="tarea">
            <button class="btn btn-outline-secondary" type="button" @click="addTareaStore" >Agregar</button>
        </div>
        <div class="row my-5">
            <div class="col-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 my-3" v-for="(item,index) in enviarDatosApi" :key="index">
                <div class="card" :class="{sombras : item.favorito}">
                    <img :src="item.avatar" class="card-img-top" :alt="item.id">
                    <div class="card-body">
                        <h5 class="card-title">{{item.first_name}} {{item.last_name}}</h5>
                        <p class="card-text">{{item.email}}</p>
                        <button class="btn btn-primary" @click="agregandoFavoritos(item)" :disabled="item.favorito">Agregar a Favoritos</button>
                    </div>
                </div>
            </div>
        </div>
    </div>
</template>

<script>
import { mapGetters, mapActions } from "vuex";

export default {
    name: 'AddTareas',
    data() {
        return {
            tarea: ''
        }
    },
    computed: {
        ...mapGetters(['enviarDatosApi'])
    },
    methods: {
        addTareaStore(){
            if (this.tarea) {
                this.$store.dispatch('agregarTarea',this.tarea)
            } else {
                console.log("no se puede");
            }
        },
        ...mapActions(['agregandoFavoritos'])
    },
}
</script>

<style scoped>
    img {
        max-height: 150px;
    }
    .sombras{
        border: 2px solid red;
        background: rgb(2,0,36);
        background: linear-gradient(90deg, rgba(2,0,36,1) 0%, rgba(132,132,201,1) 48%, rgba(0,212,255,1) 100%); 
    }
</style>