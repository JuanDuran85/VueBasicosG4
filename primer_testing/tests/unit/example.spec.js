import { shallowMount } from '@vue/test-utils';
import AddTareas from '@/components/AddTareas.vue';

describe('Prueba al componente AddTareas.vue', () => {
  it('prueba N°1', () => {
    const wrapper = shallowMount(AddTareas);

    // probar si esta montado el componente
    expect(wrapper.vm._isMounted).toBe(true);

    // probar la data del componente
    const datosTareas = AddTareas.data();
    expect(datosTareas.tarea).toEqual("");
    //expect(datosTareas.tareas).toEqual(""); genera error

    // encontrar elemento del DOM y probar sus atributos propios
    let inputTarea = wrapper.find("input");
    expect(inputTarea.element.placeholder).toEqual("Agrega una actividad");
    expect(inputTarea.element.type).toEqual("text");

    let entrada = `<input type="text" placeholder="Agrega una actividad" aria-label="Agrega una actividad" aria-describedby="button-addon2" class="form-control">`;
    expect(wrapper.find('.input-group').element.firstChild.outerHTML).toBe(entrada);

/*   expect(wrapper.find('.input-group').element.firstChild.outerHTML).toBe('<input type=\"text\" placeholder=\"Agrega una actividad\" aria-label=\"Agrega una actividad\" aria-describedby=\"button-addon2\" class=\"form-control\">'); */
  });

  it('prueba N°2', () => {
    const wrapper = shallowMount(AddTareas);

    // encontrando elementos en el DOM
    expect(wrapper.find("button").text()).toBe("Agregar");

    // probando data del componente
    let tareaNueva = "";
    expect(wrapper.vm.$data.tarea).toBe(tareaNueva);

    // usando funciones de jest
    wrapper.vm.addTareaStore = jest.fn();
    wrapper.vm.addTareaStore();
    expect(wrapper.vm.addTareaStore.mock.calls.length).toBe(1);
  });

  it('prueba N°3', ()=>{
    const wrapper = shallowMount(AddTareas);
    let inputEntrada = wrapper.find('input');
    inputEntrada.setValue("Nueva tarea numero 2");
    expect(wrapper.vm.$data.tarea).toBe("Nueva tarea numero 2");
    expect(wrapper.find('.btn').text().length).toBe(7);
    expect(wrapper.element).toMatchSnapshot();
  });
})
