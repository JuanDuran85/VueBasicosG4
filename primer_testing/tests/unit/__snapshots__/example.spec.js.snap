// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba al componente AddTareas.vue prueba NÂ°3 1`] = `
<div
  class="container my-5"
>
  <h2
    class="text-center my-4"
  >
    Agrega una Actividad por Realizar
  </h2>
   
  <div
    class="input-group mb-3"
  >
    <input
      aria-describedby="button-addon2"
      aria-label="Agrega una actividad"
      class="form-control"
      placeholder="Agrega una actividad"
      type="text"
    />
     
    <button
      class="btn btn-outline-secondary"
      type="button"
    >
      Agregar
    </button>
  </div>
</div>
`;
